Show notices when WP or PHP versions are not met

Aljullu opened the pull request:
While reviewing #1157 I noticed we are not showing any feedback to the user when PHP or WP dependencies are not met. This PR adapts the [Jetpack version check](https://github.com/Automattic/jetpack/blob/master/jetpack.php) to WooCommerce Blocks so we display a notice when minimum requirements are not met.

### Screenshots
![image](https://user-images.githubusercontent.com/3616980/68659320-5378fe80-0537-11ea-9150-9860dd070625.png)

### How to test the changes in this Pull Request:

1. In a WP 4.9 install, try to activate WooCommerce Blocks and verify a notice appears.
2. Same for a install with PHP < 5.6.
3. Alternatively, you can change the values of `WOOCOMMERCE_BLOCKS__MINIMUM_WP_VERSION` and `WOOCOMMERCE_BLOCKS__MINIMUM_PHP_VERSION` to something higher than your current install to trigger the notices.

### Changelog

> Show error notices when WP or PHP minimum versions are not met

nerrad commented on woocommerce-gutenberg-products-block.php:
```diff
@@ -18,7 +18,72 @@
+				<?php
+				printf(
+					'<a class="button button-primary" href="%1$s" target="_blank" rel="noopener noreferrer">%2$s <span class="screen-reader-text">%3$s</span><span aria-hidden="true" class="dashicons dashicons-external"></span></a>',
+					esc_url( wp_get_update_php_url() ),
```
This function was added to WordPress in version 5.1, so it won't be available if the plugin is in a WordPress 5.0 environment and has something less than PHP 5.6.  So you can't use this function here if this notice stays.

Aljullu replied:
Good catch... I think we can remove that button completely. Done in 0ec623e032452e58d71dff16ced750695c948ebb.

mikejolley commented:
> currently when bundled with Woo Core, will this notice show? Should it? Should the wording be different depending on the context (bundled with Woo Core vs installed as a plugin separately).

No, notices should not be shown when part of Woo Core. Woo core has it's own notices and wording. This should be feature plugin only (so keeping this in the main file is fine).

mikejolley commented on woocommerce-gutenberg-products-block.php:
```diff
@@ -18,7 +18,72 @@
+
+if ( version_compare( $GLOBALS['wp_version'], WOOCOMMERCE_BLOCKS__MINIMUM_WP_VERSION, '<' ) ) {
+	if ( defined( 'WP_DEBUG' ) && WP_DEBUG ) {
+		error_log( // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
```
Why do we need error logs for this? I agree with admin notice but I don't think we should clutter up logs if this is not running, as it would trigger every page load for all users..

Aljullu replied:
Removed in 5f86e63d3668c6b38c1169fd6d4ca3fa85c77b2b.

nerrad commented:
> No, notices should not be shown when part of Woo Core. Woo core has it's own notices and wording. This should be feature plugin only (so keeping this in the main file is fine).

👍  I wrote my comments before I understood how the loading process works in Woo Core (where the main file isn't even touched :) ).

Aljullu commented on woocommerce-gutenberg-products-block.php:
```diff
@@ -18,7 +18,72 @@
 
-if ( version_compare( PHP_VERSION, '5.6.0', '<' ) ) {
+define( 'WOOCOMMERCE_BLOCKS__MINIMUM_WP_VERSION', '5.0' );
+define( 'WOOCOMMERCE_BLOCKS__MINIMUM_PHP_VERSION', '5.6' );
```
Done in d2722e45215ff535f7be7e007f5182abe25c4f13.

Aljullu commented:
Thanks for the reviews @nerrad & @mikejolley. I addressed most of the feedback so feel free to take another look.

@nerrad:

> related to the above, if a notice is left in place, context for the notice matters and I'm not sure we should be showing this notice on every route in the admin. Should it be specific to the plugins route and maybe woocommerce settings pages route?

Makes sense! Done in 96e79b2a773299f01fb78e752eaa63ae5143b51c.

> I notice that the minimum WC version required for blocks is 3.6. Since WC 3.6 had a minimum WP version requirement of 5.2.4, and WC 3.7 was bumped to require 5.6, I wonder if we should just bump our minimum WC version requirements to WC 5.6 instead?

Now that WooCommerce 3.8 has been released, that makes sense to me. @mikejolley how does it sound bumping the minimum WC version to 3.7?

Aljullu commented:
> yes I think we should bump our min requirements

Ok, will do that (and remove the PHP notice that will be no longer needed) in a follow-up, so we keep PRs independent of each other.
